api.go

```diff
@@ -278,6 +283,8 @@ func privateNew(endpoint string, opts *Options) (*Client, error) {
 	// healthcheck is not initialized
 	clnt.healthStatus = unknown
 
+	clnt.maxRetries = opts.MaxRetries
```

Comment:
If MaxRetries is set to 0 set it to 3 to retain the current behavior, always default to MaxRetry global value

Reply:
i was checking it [here](https://github.com/minio/minio-go/pull/2013/files#diff-f7bb71ca856c07b3329265f26b1e577e00b71e4792550cf14346bdbe0e7bd18dR604) before, but your suggestion is cleaner. updated in https://github.com/minio/minio-go/pull/2013/commits/fef61933ab1f75afb9b1b02a6d58ab1912fc4763